### DISCUSSION TITLE
Adding more examples to supporting infrastructure

### DISCUSSION THREAD

**Discussion by navveenb** [Green-Software-Foundation/sci#93] [General]
In Methodology Summary on the specification, few things we can add explicitly -

1. How do we capture the carbon footprint of entire software development and deployment process -  For instance environments for development , Q/A, pre-production should be considered, along with the final carbon footprint of production application.
 
2. Data acquisition process - How data is acquired -APIs, feeds, crawling content, documents, social media etc ( the compute instances and network required for data acquisition )

3. Big Data infrastructure / Data Lakes - Calling out this explicitly might help for users to understand what should be covered as part of supporting infrastructure , as this is usually a big part of many enterprises.

**Comment by atg-abhishek** [comment]
@navveenb would this be too prescriptive? I'm wondering if as a part of the standard, we can make such specific recommendations?

**Reply by navveenb** [reply]
Thanks @atg-abhishek  for the input.

I am sharing my views below -
These are specific scenarios and examples that might be helpful to call out explicitly. 

Like for developing a SaaS or any digital application, should you consider
the carbon emission required to develop the application. If yes, this includes emissions from various environments - development , pre-prod etc. The carbon emission of developing applications and incremental updates should be considered as part of the software lifecycle and for estimation purpose. 

The other are role specific - the thought process was that every role has a part to play in reducing carbon emissions and including them as stakeholders in the specification - like managing a big data infrastructure and emissions associated with storing and capturing data. I mentioned big data, as we are talking about TBs of data that has been accumulated for analytics, documents, training models etc and hence this might have the largest carbon footprint.

On a separate topic, also from an awareness/education perspective, while cloud (or any system) provides infinite resources, there is a need to look at how to use various services and storage responsibly from a carbon emission perspective. This might give some direction for the various roles and be involved in the process.

**Reply by atg-abhishek** [reply]
I feel that if we keep it high-level, we can add that into the SCI directly so that it is not too prescriptive and then provide some more concrete recommendations that take into account some of the use cases that we have and include that on the [wiki](https://github.com/Green-Software-Foundation/software_carbon_intensity/wiki) page?

**Comment by jawache** [comment]
I think some of those statements are around what to include in the software boundary? If so @navveenb are they already covered in the existing statement regarding [software boundary](https://github.com/Green-Software-Foundation/software_carbon_intensity/blob/dev/Software_Carbon_Intensity/Software_Carbon_Intensity_Specification.md#software-boundary), I already see references there to build and deploy pipelines, testing, scanning. I don't think we need to be exhaustive but definitely illustrative.

But generally, I do agree with the sentiment that there needs to be more information about how to apply the SCI in the real world. @navveenb I was thinking it would be great to launch the SCI with several articles talking about how to apply the SCI in different use cases. One 1000 word article about each use case, "How to calculate the software carbon intensity of an ML model?" I don't think we need to go into too much depth, no need to calculate the numbers, just what you would include, which data you would use, what granularity you would aim for. What do you think @navveenb ?

**Reply by navveenb** [reply]
Sure @jawache , I can take this up and write articles around various real-world use cases.

**Reply by navveenb** [reply]
Created the use case at - https://github.com/Green-Software-Foundation/software_carbon_intensity/discussions/120 .
